v22.1/alter-backup.md

```diff
@@ -0,0 +1,95 @@
+---
+title: ALTER BACKUP
+summary: Use the ALTER BACKUP statement to add `NEW_KMS` encryption keys to backups.
+toc: true
+docs_area: reference.sql
+---
+
+{% include enterprise-feature.md %}
+
+<span class="version-tag">New in v22.1:</span> The `ALTER BACKUP` statement allows for new KMS encryption keys to be applied to existing encrypted backups. Each `ALTER BACKUP` statement must include the new KMS encryption key with `NEW_KMS`, and use `WITH OLD_KMS` to refer to at least one of the KMS URIs that were originally used to encrypt the backup.
+
+After an `ALTER BACKUP` statement successfully completes, subsequent `BACKUP`, `RESTORE`, and `SHOW BACKUP` statements can use any of the existing or new KMS URIs to decrypt the backup.
+
+CockroachDB supports AWS and Google Cloud KMS keys. For more detail on encrypted backups and restores, see [Take and Restore Encrypted Backups](take-and-restore-encrypted-backups.html).
+
+## Synopsis
+
+<div>
+{% remote_include https://raw.githubusercontent.com/cockroachdb/generated-diagrams/master/grammar_svg/alter_backup.html %}
+</div>
+
+## Parameters
+
+Parameter         | Description
+------------------+-------------------------------------------------------------------------------------------------------------------------
+`subdirectory`    | The subdirectory that contains the target backup at the given `collectionURI`.
+`LATEST`          | The most recent backup at the given `collectionURI`.
+`collectionURI`   | The URI that holds the backup collection.
+`ADD NEW_KMS`     | Apply the new KMS encryption key to the target backup.
+`WITH OLD_KMS`    | Reference one of the existing KMS URI(s) originally used to encrypt the backup.
+`kmsURI`          | The [URI](take-and-restore-encrypted-backups.html#uri-formats) for the KMS key.
+
+## Required privileges
+
+- `ALTER BACKUP` can only be run by members of the [`admin` role](security-reference/authorization.html#admin-role). By default, the `root` user belongs to the `admin` role.
+- `ALTER BACKUP` requires full read and write (including delete and overwrite) permissions to the target cloud storage bucket.
```

Comment:
Does it need delete and overwrite? ALTER BACKUP should write new encryption files side by side so I don't see why it would need to delete or overwrite ever, but maybe I'm missing something?

Reply:
Yes, it sounds like you're right! This is the language from the _current_ status of `BACKUP` (soon-to-change)... but yes, since we're **adding** encryption keys and not removing in any way I assume that delete and overwrite can be removed?

Reply:
Yeah that seems right to me.

And I believe what we discussed is that even if we wanted to remove encryption keys we still shouldn't overwrite or delete as we just don't want to do that period for backups.